docs/usage/json_schema.md

```diff
@@ -826,4 +716,67 @@ print(json.dumps(Person.model_json_schema(), indent=2))
 """
 ```
 
-Note that you *must* return a schema, even if you are just mutating it in-place.
+Note that you *must* return a schema, even if you are just mutating it in place.
+
+## Customizing the JSON schema generation process
+
+If you need custom schema generation, you can use a `schema_generator`, modifying the
+[`GenerateJsonSchema`][pydantic.json_schema.GenerateJsonSchema] class as necessary for your application.
```

Comment:
@dmontagu I defer to you if you want to document this, or document it with a big warning at the top saying it's experimental?

Reply:
I think let's  hold off on documenting for now but plan to include it with a warning when we add it, maybe shortly. I'd like to give FastAPI a bit of time first.